wal/src/rocks_impl/manager.rs

```diff
@@ -270,40 +270,114 @@ impl RocksImpl {
                 delete_lock: Mutex::new(()),
             };
 
-            let location = WalLocation {
-                versioned_region_id,
-                table_id,
-            };
-            table_units.insert(location, Arc::new(table_unit));
+            table_units.insert(table_id, Arc::new(table_unit));
         }
 
         Ok(())
     }
 
     fn find_table_seqs_from_table_log(
+        &self,
+        table_max_seqs: &mut HashMap<TableId, SequenceNumber>,
+    ) -> Result<()> {
+        self.find_table_seqs_from_table_log_by_region_id(table_max_seqs)
+    }
+
+    fn find_table_seqs_from_table_log_by_region_id(
+        &self,
+        table_max_seqs: &mut HashMap<TableId, SequenceNumber>,
+    ) -> Result<()> {
+        let mut current_region_id = RegionId::MAX;
+        let mut end_boundary_key_buf = BytesMut::new();
+        loop {
+            debug!(
+                "RocksImpl searches table logs for sequences by region id, region id:{}",
+                current_region_id
+            );
+
+            let log_key = CommonLogKey::new(current_region_id, TableId::MAX, MAX_SEQUENCE_NUMBER);
+            self.log_encoding
+                .encode_key(&mut end_boundary_key_buf, &log_key)
+                .map_err(|e| Box::new(e) as _)
+                .context(Encoding)?;
+            let mut iter = self.db.iter();
+            let seek_key = SeekKey::Key(&end_boundary_key_buf);
+
+            let found = iter
+                .seek_for_prev(seek_key)
+                .map_err(|e| e.into())
+                .context(Initialization)?;
+
+            if !found {
+                debug!(
+                    "RocksImpl find table unit pairs stop scanning by region id, because of no entries to scan."
+                );
+                break;
+            }
+
+            if !self
+                .log_encoding
+                .is_log_key(iter.key())
+                .map_err(|e| Box::new(e) as _)
+                .context(Decoding)?
+            {
+                debug!(
+                    "RocksImpl find table unit pairs stop scanning by region id, because log keys are exhausted."
+                );
+                break;
+            }
+
+            let log_key = self
+                .log_encoding
+                .decode_key(iter.key())
+                .map_err(|e| Box::new(e) as _)
+                .context(Decoding)?;
+
+            // Found a valid log key by `current_region_id`, search table sequences by table
```

Comment:
Actually, the log key may not belong to `current_region_id`, let's put the comment as:
```suggestion
            // The max valid log key in a region is found, search table sequences by table then.
```